packages/ra-core/src/dataProvider/withLifecycleCallbacks.ts

```diff
@@ -290,42 +268,23 @@ export const withLifecycleCallbacks = (
             params: UpdateParams<RecordType>
         ) {
             let newParams = params;
-            const beforeUpdateHandlers = handlers.filter(
-                h => h.resource === resource && h.beforeUpdate
-            );
-            for (let handler of beforeUpdateHandlers) {
-                newParams = await handler.beforeUpdate(newParams, dataProvider);
-            }
-            const beforeSaveHandlers = handlers.filter(
-                h => h.resource === resource && h.beforeSave
-            );
-            for (let handler of beforeSaveHandlers) {
-                newParams.data = await handler.beforeSave(
-                    newParams.data,
-                    dataProvider
-                );
-            }
 
-            let result = await dataProvider.update<RecordType>(
+            newParams = await applyCallbacks(
                 resource,
+                'beforeUpdate',
                 newParams
             );
-
-            const afterUpdateHandlers = handlers.filter(
-                h => h.resource === resource && h.afterUpdate
+            newParams = await applyCallbacks(resource, 'beforeSave', newParams);
```

Comment:
This is another breaking change: beforeSave used to be applied on params.data, not on params

Reply:
changed